src/ILCompiler/src/RemovingILProvider.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+using Internal.IL;
+using Internal.IL.Stubs;
+using Internal.TypeSystem;
+
+namespace ILCompiler
+{
+    public class RemovingILProvider : ILProvider
+    {
+        private readonly ILProvider _baseILProvider;
+        private readonly RemovedFeature _removedFeature;
+
+        public RemovingILProvider(ILProvider baseILProvider, RemovedFeature feature)
+        {
+            _baseILProvider = baseILProvider;
+            _removedFeature = feature;
+        }
+
+        public override MethodIL GetMethodIL(MethodDesc method)
+        {
+            switch (GetAction(method))
+            {
+                case RemoveAction.Nothing:
+                    return _baseILProvider.GetMethodIL(method);
+
+                case RemoveAction.ConvertToStub:
+                    if (method.Signature.ReturnType.IsVoid)
+                        return new ILStubMethodIL(method, new byte[] { (byte)ILOpcode.ret }, Array.Empty<LocalVariableDefinition>(), null);
+                    else if (method.Signature.ReturnType.Category == TypeFlags.Boolean)
+                        return new ILStubMethodIL(method, new byte[] { (byte)ILOpcode.ldc_i4_0, (byte)ILOpcode.ret }, Array.Empty<LocalVariableDefinition>(), null);
+                    else
+                        goto default;
+
+                case RemoveAction.ConvertToTrueStub:
+                    return new ILStubMethodIL(method, new byte[] { (byte)ILOpcode.ldc_i4_1, (byte)ILOpcode.ret }, Array.Empty<LocalVariableDefinition>(), null);
+
+                case RemoveAction.ConvertToGetResourceStringStub:
+                    return new ILStubMethodIL(method, new byte[] {
+                        (byte)ILOpcode.ldarg_0,
+                        (byte)ILOpcode.ret },
+                        Array.Empty<LocalVariableDefinition>(), null);
+
+                case RemoveAction.ConvertToThrow:
+                    return new ILStubMethodIL(method, new byte[] { (byte)ILOpcode.ldnull, (byte)ILOpcode.throw_ }, Array.Empty<LocalVariableDefinition>(), null);
+
+                default:
+                    throw new NotImplementedException();
+            }
+        }
+
+        private RemoveAction GetAction(MethodDesc method)
+        {
+            TypeDesc owningType = method.OwningType;
+
+            if (!method.Signature.IsStatic)
```

Comment:
Nit: It may be more readable to structure it like this:

```
if ((_removedFeature & RemovedFeature.Etw) != 0)
{
     ... logic specific to ETW ...    
}

if ((_removedFeature & RemovedFeature.FrameworkResources) != 0)
{
     ... logic specific to framework resources ...    
}
```